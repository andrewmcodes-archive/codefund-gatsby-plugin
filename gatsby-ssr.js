"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = require("react");
exports.onRenderBody = function (_a, config) {
    var setHeadComponents = _a.setHeadComponents;
    if (process.env.NODE_ENV === "production") {
        return setHeadComponents([
            react_1.createElement("script", {
                key: "codefund-gatsby-plugin",
                src: "https://app.codefund.io/properties/" + config.propertyId + "/funder.js",
                async: "async",
            }),
        ]);
    }
};
