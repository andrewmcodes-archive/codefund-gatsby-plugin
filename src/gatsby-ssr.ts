import { createElement } from "react"
import { GatsbySsrContext } from "./types/gatsby"
import { PluginConfig } from "./gatsby-node"

export const onRenderBody = (
  { setHeadComponents }: GatsbySsrContext,
  config: PluginConfig
) => {
  if (process.env.NODE_ENV === `production`) {
    return setHeadComponents([
      createElement("script", {
        key: `codefund-gatsby-plugin`,
        src: `https://app.codefund.io/properties/${config.propertyId}/funder.js`,
        async: `async`,
      }),
    ])
  }
}
