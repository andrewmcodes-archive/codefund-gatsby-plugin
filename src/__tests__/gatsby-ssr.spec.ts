"use strict"
import { onRenderBody } from "../gatsby-ssr"

describe(`@codefund/gatsby-plugin`, () => {
  describe(`onRenderBody`, () => {
    describe(`in development mode`, () => {
      it(`does not set any head components`, () => {
        const setHeadComponents = jest.fn()
        const config = { propertyId: `0` }
        onRenderBody({ setHeadComponents }, config)

        expect(setHeadComponents).not.toHaveBeenCalled()
      })
    })

    describe(`in production mode`, () => {
      let env: any

      beforeAll(() => {
        env = process.env.NODE_ENV
        process.env.NODE_ENV = `production`
      })

      afterAll(() => {
        process.env.NODE_ENV = env
      })

      it(`sets the correct head components`, () => {
        const setHeadComponents = jest.fn()
        const config = { propertyId: `1` }

        onRenderBody({ setHeadComponents }, config)

        expect(setHeadComponents.mock.calls).toMatchSnapshot()
      })
    })
  })
})
