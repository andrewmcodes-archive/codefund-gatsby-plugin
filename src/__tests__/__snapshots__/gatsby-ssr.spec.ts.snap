// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@codefund/gatsby-plugin onRenderBody in production mode sets the correct head components 1`] = `
Array [
  Array [
    Array [
      <script
        async="async"
        src="https://app.codefund.io/properties/1/funder.js"
      />,
    ],
  ],
]
`;
